### PR TITLE
Authpages

### PR DESCRIPTION
**Pull request by Eyiowuawi**
this branch contains the following:
1) login, signup, new password and forgot password pages 
2) product listing page
3) bookmark context & bookmark functionality

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,14 +4,18 @@ import './index.css';
...
 
 
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
+      
     <BrowserRouter>
-      <App />
+      <Context>
```
Comment (mayowadabiri) [inline src/index.js:16]: This context is used by what component?? What is work of the context?

**Review by mayowadabiri — CHANGES_REQUESTED** [review]
- You need to work on your naming convention
- I don't think the context you're using is needed at this point.. Context API + useReducer is used for managing complex state. 

Nice work by the way

File: src/Context/Context.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { createContext, useReducer, useContext } from 'react';
+import { bookmarkReducer } from './Reducer';
+import Array from '../Constant/Array';
+
+const Bookmark = createContext();
```
[the remark this answers is not in the document]
Comment (Eyiowuawi) [reply src/Context/Context.jsx:5]: corrected

File: src/Constant/Array.jsx [hunk 3]
```diff
@@ -61,5 +74,113 @@ export const Array = [
     BedroomNumber: 2,
     BathroomNumber: 2,
   },
+  {
+    id: 7,
+    image: House7,
+    Price: '10,000',
```
[the remark this answers is not in the document]
Comment (Eyiowuawi) [reply src/Constant/Array.jsx:80]: done and i have also check W3schools for more insight.. thanks sir

File: src/Context/Context.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+const Context = ({ children }) => {
+  const [state, dispatch] = useReducer(bookmarkReducer, {
+    apartments: Array,
```
[the remark this answers is not in the document]
Comment (Eyiowuawi) [reply src/Context/Context.jsx:9]: this has also been changed to appartments

File: src/components/ApartmentCard/ApartmentCard.jsx [hunk 5]
```diff
@@ -2,9 +2,23 @@ import React from 'react';
...
+import { BookmarkState } from '../../Context/Context';
 
-const ApartmentCard= ({ item }) => {
+const ApartmentCard = ({ item }) => {
+  const {
+    state: { bookmark },
+    dispatch,
+  } = BookmarkState();
+  const add = () => {
+    console.log(bookmark);
```
[the remark this answers is not in the document]
Comment (Eyiowuawi) [reply src/components/ApartmentCard/ApartmentCard.jsx:18]: done

File: src/components/Button/Button2.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import {Link} from 'react-router-dom'
+import styles from './Button.module.css';
+import backarrow from '../../Assets/backarrow.png';
+
+function Button2({ children }) {
```
[the remark this answers is not in the document]
Comment (Eyiowuawi) [reply src/components/Button/Button2.jsx:6]: done

File: src/components/Button/Button2.jsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    return (
+    <Link to='/'>
```
[the remark this answers is not in the document]
Comment (Eyiowuawi) [reply src/components/Button/Button2.jsx:8]: it's the button component that can take image... even tho currently the design is up for change to just contain an image (back arrow).

File: src/index.js (hunk 1 above)
Comment (Eyiowuawi) [reply src/index.js:16]: the context is used by the Featured apartment on the landing page, the apartment listing page and the bookmark page which might have an icon that tracks the length of bookmarked apartment array on the nav bar just like the cart length in my last project

File: src/components/FeaturedApartment/FeaturedApartment.js [hunk 8]
```diff
@@ -1,22 +1,28 @@
 import React from 'react';
 import styles from './FeauturedApartment.module.css'
-import Array from '../../Constant/Array';
 import ApartmentCard from '../ApartmentCard/ApartmentCard';
 import Button from './../Button/Button';
+import {BookmarkState} from '../../Context/Context';
```
Comment (mayowadabiri) [inline src/components/FeaturedApartment/FeaturedApartment.js:5]: Change this to small letter. `bookmarkState`. Also, change the file name to small letter

File: src/index.js [hunk 9]
```diff
@@ -4,14 +4,18 @@ import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
 import { BrowserRouter } from "react-router-dom";
+import Context from './Context/Context';
```
Comment (mayowadabiri) [inline src/index.js:7]: Here also

File: src/pages/Apartmentlist/Apartmentlist.jsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import styles from './Apartmentlist.module.css'
+// import ApartmentCard from '../ApartmentCard/ApartmentCard';
+import ApartmentCard from '../../components/ApartmentCard/ApartmentCard';
```
Comment (mayowadabiri) [inline src/pages/Apartmentlist/Apartmentlist.jsx:4]: Can you check the way I am importing files now.. Check the set of files I merged, like Aisha's file.. It's imported as just `'components/ApartmentCard/ApartmentCard'` No need for those back slashes

File: src/pages/ResetPassword/ResetPassword.jsx [hunk 11]
```diff
@@ -1,5 +1,24 @@
 import React from 'react';
+import FormGroup from './../../components/FormGroup/FormGroup';
```
Comment (mayowadabiri) [inline src/pages/ResetPassword/ResetPassword.jsx:2]: Pls work on the way you import files

**Review by mayowadabiri — CHANGES_REQUESTED** [review]

File: src/components/Button/Button2.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (mayowadabiri) [reply src/components/Button/Button2.jsx:8]: Why using a and button tag together.. That's semantically not correct.
Is it leading to another page to going to perform an action?

**Pull request merged**